src/wsdiscovery.ts

```diff
@@ -213,44 +306,59 @@ export default class WsDiscovery extends EventEmitter {
                         pid: this.networkDevicePID(name),
                     };
                     const device = new HuddlyDevice(deviceData);
-
-                    if (this.linkLocalAddrAllowed(device.ip.toString())) {
+                    if (this.isDeviceAllowed(device.ip.toString())) {
                         discoveredDevices.push(device);
                         this.emit('device', device);
                     } else {
                         Logger.debug(
-                            `Link local huddly ip camera detected. Instructions are set to ignore link local devices!`,
+                            `Ignoring device at ip address: ${device.ip.toString()}.`,
+                            WsDiscovery.name
+                        );
+                        Logger.debug(
+                            `Ignore local link devices: ${this.opts.ignoreLinkLocalDevices ||
+                                process.env.HUDDLY_WSDD_IGNORE_LINK_LOCAL_DEVICES}.`,
+                            WsDiscovery.name
+                        );
+                        Logger.debug(
+                            `Has full probing rights: ${this.opts.probeEntireNetwork ||
+                                process.env.PROBE_ENTIRE_NETWORK}`,
                             WsDiscovery.name
                         );
                     }
                 });
             }
             callback(discoveredDevices);
         };
-        this.socket.on('message', onProbeResponseHandler);
-        setTimeout(() => {
-            self.socket.removeListener('message', onProbeResponseHandler);
-            callback(discoveredDevices);
-        }, this.opts.timeout);
 
-        if (this.ifStateConnected) {
+        Object.values(this.socketConnections).forEach(socket => {
+            socket.on('message', onProbeResponseHandler);
+            setTimeout(() => {
+                socket.removeListener('message', onProbeResponseHandler);
+            }, this.opts.timeout);
+
             this.setTimeoutWithRandomDelay(
-                this.socket.send.bind(this.socket, body, 0, body.length, 3702, '239.255.255.250'),
+                socket.send.bind(socket, body, 0, body.length, 3702, '239.255.255.250'),
                 this.maxDelay
             );
```

Comment:
A bit concerned with the timeout intervals within the foreach loop. When it goes out of context, does it actually call the `socket.removeListener` when the socket does not get a message?

Reply:
I will test it. Good thikning!

Reply:
Doesn't seem to be an issue removing the listener at the timeout intervals :)